pennylane_qiskit/converter.py

```diff
@@ -36,17 +36,83 @@
 dagger_map = {"SdgGate": qml.S, "TdgGate": qml.T, "SXdgGate": qml.SX}
 
 
-def _check_parameter_bound(param: Parameter, var_ref_map: Dict[Parameter, Any]):
+def _check_parameter_bound(param: Parameter, trainable_params: Dict[Parameter, Any]):
     """Utility function determining if a certain parameter in a QuantumCircuit has
     been bound.
 
     Args:
         param (qiskit.circuit.Parameter): the parameter to be checked
-        var_ref_map (dict[qiskit.circuit.Parameter, Any]):
+        trainable_params (dict[qiskit.circuit.Parameter, Any]):
             a dictionary mapping qiskit parameters to trainable parameter values
     """
-    if isinstance(param, Parameter) and param not in var_ref_map:
-        raise ValueError(f"The parameter {param} was not bound correctly.")
+    if isinstance(param, Parameter) and param not in trainable_params:
+        raise ValueError(f"The parameter {param} was not bound correctly.".format(param))
+
+
+def _format_params_dict(quantum_circuit, params, *args, **kwargs):
+    """Processes the inputs for calling the quantum function and returns
+    a dictionary of the format ``{Parameter("name"): value}`` for all the parameters.
+
+    For a ``quantum_circuit`` with parameters ``[Parameter("phi"), Parameter("psi"), Parameter("theta")]``,
+    inputs can be one of the following:
+        1. the kwargs passed to when calling the qfunc, other than ``params`` and ``wires``.
+            The keys in kwargs are expected to correspond to the names of the Parameters on
+            the QuantumCircuit, i.e. ``(qc, None, phi=0.35, theta=0.2, psi=1.7)``
+        2. the args passed when calling the qfunc, i.e. ``(qc, None, 0.35, 1.7, 0.2)``. These
+           are assigned to the Parameters as the are ordered on the Qiskit QuantumCircuit:
+           alphabetized by Parameter name
+        3. Some combination of args and kwargs, i.e. ``(qc, None, 0.35, 0.2, psi=1.7)``
+        4. (legacy) ``params`` from the kwarg ``params`` of the qfunc call, which is expected
+            to already be a dictionary of the format ``{Parameter("name"): value}``, i.e.
+            ``(qc, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})``
+        5. (legacy) ``params`` passed as a single arg, which is expected
+            to already be a dictionary of the format ``{Parameter("name"): value}``, i.e.
+            ``(qc, None, {Parameter("phi"): 0.35, Parameter("psi"): 1.7, Parameter("theta"): 0.2})``
+
+    Returns:
+        params (dict): A dictionary mapping ``quantum_circuit.parameters`` to values
+    """
+
+    # if no kwargs are passed, and a dictionary has been passed as a single argument, then assume it is params
+    if params is None and not kwargs and (len(args) == 1 and isinstance(args[0], dict)):
+        return args[0]
+
+    # make params dict if using args and/or kwargs
+    if args or kwargs:
```

Comment:
Could potentially use a guard-clause here and reduce the indentation of this next block:
```
if not args and not kwargs:
    return params
if params is not None:
```